### PR TITLE
Add ctrlf-mode-map

### LINKED ISSUE #62: Make minibuffer keymap available
Hi,

I like using [which-key](https://github.com/justbur/emacs-which-key/) to inspect keymaps. Is there a reason why the keymap itself has to be created on the fly? Can it be the global variable we get to customize instead of the list of bindings.

### PR DESCRIPTION
**Pull request by rgrinberg**
Attempt to fix #62.

I'm not sure I fully understand your
[comment](https://github.com/raxod502/ctrlf/issues/62#issuecomment-711087456)(my
elisp is weak), but I figure it's easier to continue the discussion
with some concrete code.

----

#

### CODE REVIEW

**Review by raxod502 — COMMENTED** [review]
Yeah, this is exactly the check that I was suggesting. Thanks for taking it on! Just a few small changes:
* I think we now need to rename the reference to `ctrlf--keymap` below to `ctrlf-mode-map`.
* I think we probably want to keep the `(setcdr ctrlf-mode-map nil)` call (only in the case where your `when` clause passes, of course). It makes it so if you remove a binding from `ctrlf-mode-map`, it actually gets deactivated.
* We should update the docstring of `ctrlf-mode-bindings` to say it's deprecated, and probably mark the variable as such (I think this is done with `make-obsolete-variable`, or perhaps `defcustom` has an option specially for this).
* There should be an accompanying README update, and changelog entry.

**Comment by raxod502** [comment]
Oh yes, also the CI is currently failing (due to the reference to `ctrlf--keymap`).

**Comment by rgrinberg** [comment]
Thanks for the review. I think I addressed all your comments.

**Comment by rgrinberg** [comment]
I can make an analogous change for `ctrlf-minibuffer-bindings`. Should it be called `ctrlf-minibuffer-map`?

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ The format is based on [Keep a Changelog].
 
 [#59]: https://github.com/raxod502/ctrlf/issues/59
 
+* Customize keybindings by modifying the mode map `ctrlf-mode-map`
+  instead of the `ctrlf-mode-bindings` variable. `ctrlf-mode-bindings`
+  is obsolete but remains available.
+
+[#64] https://github.com/raxod502/ctrlf/pull/64
+
```
Comment (raxod502) [inline CHANGELOG.md:47]: Would you mind putting the bullet points next to each other, and the link anchors together at the bottom?

File: ctrlf.el [hunk 2]
```diff
@@ -128,11 +131,12 @@ These bindings are available globally in Emacs. See also
...
-          :value-type function)
-  :set (lambda (var val)
-         (set var val)
-         (when (bound-and-true-p ctrlf-mode)
-           (ctrlf-mode +1))))
+          :value-type function))
+:set (lambda (var val)
+          (set var val)
+          (when (bound-and-true-p ctrlf-mode)
+            (ctrlf-mode +1)))
```
Comment (raxod502) [inline ctrlf.el]: I think this is indented too far.

**Review by raxod502 — COMMENTED** [review]
Thanks, that looks good to me. Yes, doing the same for `ctrlf-minibuffer-bindings` and naming it `ctrlf-minibuffer-map` sounds perfect.

**Comment by rgrinberg** [comment]
@raxod502 sorry for the delay. should  be ready now

**Comment by raxod502** [comment]
Would you mind fixing the CI failures? This also may need a bit of additional consideration with the change https://github.com/raxod502/ctrlf/commit/ddfbc358b092351167219ca5884a25b2eb8bac00 from February.

**Comment by raxod502** [comment]
This thread is being closed automatically by [Tidier](https://github.com/raxod502/tidier) because it is labeled with "waiting on response" and has not seen any activity for 90 days. But don't worry—if you have any information that might advance the discussion, leave a comment and I will be happy to reopen the thread :)

**Pull request closed without merging**